lib/unique_elements.rb

```diff
@@ -2,5 +2,15 @@
 # combination of the above), and returns an array of the elements that appear once and only
 # once.
 def find_unique_elements(arr)
-  
+    #select each element of the array and compare each one to the rest of the array
+    arr.each_index do |idx|
+    #compare each element within the array
+        if arr[idx] == arr[idx+1]
+            arr.delete_at(idx)
+            #delete the first matching element
+            arr.delete_at(idx+1)
```

Comment:
There's an issue here with how you are deleting data from the Array while you are looping through it by index.

Let's take the array `[1, 2, "three", "three"]`
In your loop, when `idx` is `2`, you hit the conditional on line 8.
So first, you delete from the array at the index `2`. Now, everything else shifts to the left. Your array is:
`[1, 2, "three"]`. There is no value at index `3` to be deleted, and also now your array is only indexed up to index `2` which already ran. Your loop is finished.

A small change in _how_ you are deleting values should solve this.